src/compiler/program.ts

```diff
@@ -796,12 +809,20 @@ namespace ts {
                 }
             });
 
-            filesByName.set(fileName, file);
+            filesByName.set(normalizedAbsolutePath, file);
             if (file) {
-                skipDefaultLib = skipDefaultLib || file.hasNoDefaultLib;
+                if (host.useCaseSensitiveFileNames()) {

```

Comment:
why do not we do this all the time, instead of two different checks, something like:

```ts
let checkFileCasing = host.useCaseSensitiveFileNames() || options.forceConsistentCasingInFileNames;
 if (checkFileCasing) { 
                    const existingFile = filesByNameIgnoreCase.get(normalizedAbsolutePath);
                    if (existingFile) {
                        reportFileNamesDifferOnlyInCasingError(fileName, existingFile.fileName, refFile, refPos, refEnd);
                    }
                    else {
                        filesByNameIgnoreCase.set(normalizedAbsolutePath, file);
                    }
}
```